TDOAuth.h

```diff
@@ -28,11 +28,15 @@
 */
 
 #import <Foundation/Foundation.h>
+#import <Availability.h>
 
 
 FOUNDATION_EXPORT double TDOAuthVersionNumber;
 FOUNDATION_EXPORT const unsigned char TDOAuthVersionString[];
 
+#if defined(__IPHONE_8_0) && (__IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_8_0)
+#define USE_NSURLCOMPONENTS
+#endif
```

Comment:
Just to note. NSURLComponents were available in iOS 7 and not only in 8+.
https://developer.apple.com/documentation/foundation/nsurlcomponents?language=objc

I'm still okay raising the minimum support higher, but this gives us more room to easily support a slightly older version.

Reply:
Although NSURLComponents is available in iOS 7, queryItems is iOS 8, hence the constraint.

Reply:
That's exactly right. I went down the same path as you @dbgrandi :)